portfolio/src/main/java/com/google/sps/functionality/TextTranslator.java

```diff
@@ -0,0 +1,24 @@
+package com.google.sps.functionality;
+
+import com.google.cloud.translate.Translate;
+import com.google.cloud.translate.TranslateOptions;
+import com.google.cloud.translate.Translation;
+
+/** Class that translates text.  */
+public class TextTranslator { 
+  /** Translates `text` from original language to the language
+    * specified by `languageCode` and returns the translated text. 
+    * If `languageCode` is "none", then returns `text` unchanged.
+    */
+  public static String translateText(String text, String languageCode) {
+   if (languageCode.equals("none")) {
+      return text;
+    }
+
+    Translate translate = TranslateOptions.getDefaultInstance().getService();
+    Translation translation = translate.translate(text, 
+      Translate.TranslateOption.targetLanguage(languageCode));
```

Comment:
Do you need to set sourceLanguage?
Does your UI allow comment with a different language?

Reply:
It automatically detects the source language as far as I can tell. I have tested posting comments in different languages and then translating the comment section to English, and that seems to work.